### PR TITLE
feat(wash-lib,wash-cli): wash dev interface usage detection

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This is one of the first PRs on the way to #2508

### CODE REVIEW

File: crates/wash-cli/src/dev.rs [hunk 1]
```diff
@@ -87,6 +85,154 @@ impl Drop for HostSubprocess {
...
+    /// wasmCloud keyvalue provider (normally corresponds to `wasi:keyvalue`)
+    KeyValueProvider,
+    /// wasmCloud HTTP server provider (normally corresponds to `wasi:incoming-handler`)
+    HttpServerProvider,
+    /// wasmCloud HTTP client provider (normally corresponds to `wasi:outgoing-handler`)
+    HttpClientProvider,
+    /// wasmCloud NATS messaging provider (normally corresponds to `wasi:messaging`)
+    NatsMessagingProvider,
+    /// wasmCloud NATS messaging provider (normally corresponds to `wasi:blobstore`)
+    BlobstoreFsProvider,
```
Comment (brooksmtownsend) [inline crates/wash-cli/src/dev.rs:102]: nit: interface correct, provider just off 😄

File: crates/wash-cli/src/dev.rs [hunk 2]
```diff
@@ -87,6 +85,154 @@ impl Drop for HostSubprocess {
...
+            p.name.name,
+            iface.name.as_ref().context("interface missing name")?,
+        );
+        match name.split_once('@') {
+            Some(("wasi:keyvalue/atomics", _)) | Some(("wasi:keyvalue/store", _)) => {
+                Ok(KnownDep::KeyValueProvider)
+            }
+            _ => bail!("unsupported WIT interface [{name}]"),
+        }
+    }
```
Comment (brooksmtownsend) [inline crates/wash-cli/src/dev.rs:128]: this could also be the blobstore or NATS messaging provider right?

File: crates/wash-cli/src/dev.rs [hunk 3]
```diff
@@ -87,6 +85,154 @@ impl Drop for HostSubprocess {
...
+    fn from_wit_export_package(p: &Package, iface: &Interface) -> Result<Self> {
+        let name = &format!(
+            "{}:{}/{}",
+            p.name.namespace,
+            p.name.name,
+            iface.name.as_ref().context("interface missing name")?,
+        );
+        match name.split_once('@') {
+            Some(("wasi:http/incoming-handler", _)) => Ok(KnownDep::HttpServerProvider),
+            Some(("wasmcloud:messaging/handler", _)) => Ok(KnownDep::HttpServerProvider),
```
Comment (brooksmtownsend) [inline crates/wash-cli/src/dev.rs:140]: this should be the messaging provider right

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Looking good! Only real requests are to correct the incorrect logic, I'm fine with this being only partially filled in for now 😄

File: crates/wash-cli/src/dev.rs (hunk 2 above)
Comment (vados-cosmonic) [reply crates/wash-cli/src/dev.rs]: Sorry PR was a bit early this part wasn't all filled in yet!

File: crates/wash-cli/src/dev.rs (hunk 3 above)
Comment (vados-cosmonic) [reply crates/wash-cli/src/dev.rs]: Specifically the NATS messaging provider! PR was a bit early -- sorry the code wasn't there yet!

File: crates/wash-cli/src/dev.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/wash-cli/src/dev.rs]: Ah thanks, fixed!

**Review by brooksmtownsend — APPROVED** [review]

**Comment by brooksmtownsend** [comment]
@vados-cosmonic I removed this from the merge queue, looks like the wash unit tests + integration tests were failing?

**Comment by vados-cosmonic** [comment]
Hey @brooksmtownsend they're periodic failures due to some kind of proxy we use? See https://github.com/wasmCloud/wasmCloud/actions/runs/10704194735/job/29676580519?pr=2929

I was expecting them to *not* fail in the merge queue (the merge queue doesn't merge if the test fails right?)

**Comment by vados-cosmonic** [comment]
To save you a click:

```
thread 'start::github::test::test_download_client_with_proxy_settings' panicked at crates/wash-lib/src/start/github.rs:235:9:
assertion `left == right` failed
  left: 503
 right: 200
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Pull request merged**